### PR TITLE
Influence for sentiment analysis notebook

### LINKED ISSUE #452: Write EKFAC notebook
Include an example with a language model like GPT2 or some llama

### PR DESCRIPTION
**Pull request by Xuzzo**
### Description

This PR closes #452 

### Changes

- Adds new notebook, influence_sentiment_analysis.ipynb, which showcases ekfac for LLMs
- Fixes small issue in Ekfac implementation
- Adds methods to EkfacInfluence that return influence values for each layer
- Adds tests for new by_layer methods

### Checklist

- [x] Wrote Unit tests (if necessary)
- [x] Updated Documentation (if necessary)
- [x] Updated Changelog
- [x] If notebooks were added/changed, added boilerplate cells are tagged with `"tags": ["hide"]` or `"tags": ["hide-input"]`

### CODE REVIEW

**Comment by schroedk** [comment]
You have to add the new notebook to the `mkdocs.yml`
```
      - For sentiment analysis: examples/influence_sentiment_analysis.ipynb

```

File: src/pydvl/influence/torch/influence_function_model.py [hunk 1]
```diff
@@ -897,19 +897,22 @@ class EkfacInfluence(TorchInfluenceFunctionModel):
...
+        progress: If True, display progress bars.
     """
 
     ekfac_representation: EkfacRepresentation
 
     def __init__(
         self,
         model: nn.Module,
-        hessian_regularization: float = 0.0,
+        hessian_regularization: Optional[float] = None,
```
Comment (schroedk) [inline src/pydvl/influence/torch/influence_function_model.py:908]: This needs change back to default value 0.0

File: notebooks/influence_sentiment_analysis.ipynb [hunk 2]
```diff
@@ -0,0 +1,1328 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Found cached dataset imdb (/Users/fabio/.cache/huggingface/datasets/imdb/plain_text/1.0.0/d613c88cf8fa3bab83b4ded3713f1f74830d1100e171db75bbddb80b3345c9c0)\n",
```
Comment (schroedk) [inline notebooks/influence_sentiment_analysis.ipynb:155]: Please add "hide-output"

File: notebooks/influence_sentiment_analysis.ipynb
Comment (AnesBenmerzoug) [inline notebooks/influence_sentiment_analysis.ipynb:1]: - The imports should be sorted.
- In cell[8], you're only passing a single example so you shouldn't need padding and the attention mask.
- In cell [10], you could reuse `model_output` from cell[8]
- Also in cell[10], you should be able to access logits with `model_output.logits` instead of `model_output["logits"]`
- In cell[12], you access the logits by index instead of by attribute.
- I think the helper classes and functions in cell[14] should be moved out of the notebook. The logits wrapper may actually be useful for pyDVL users.
- In cell[17], why are the batch sizes not a power of 2? Why is the test_dataloader shuffled?
- In cell[19], there are missing words.
- Typos in cell after cell[24].

**Review by AnesBenmerzoug — APPROVED** [review]
Sorry for the late review. Overall it looks good to me.
Feel free to merge it once the comments I left are addressed.

File: notebooks/influence_sentiment_analysis.ipynb
Comment (Xuzzo) [reply notebooks/influence_sentiment_analysis.ipynb:1]: thanks! I have included your comments. Regarding the size of the batches: I have chosen them so that running the notebook would take <10 minutes on my mac and because I really wanted to have the test example with idx=4.

Comment (AnesBenmerzoug) [reply notebooks/influence_sentiment_analysis.ipynb:1]: I see. Thanks for the clarification

**Review by AnesBenmerzoug — APPROVED** [review]
